coordinator/src/node/liquidated_positions.rs

```diff
@@ -0,0 +1,533 @@
+                        orderbook::db::matches::get_matches_by_order_id(&mut conn, order.id)?;
+                    let matches: Vec<Match> = matches.into_iter().map(Match::from).collect();
+
+                    let closing_price = average_execution_price(matches)
+                        .to_f32()
+                        .expect("to fit into f32");
+                    db::positions::Position::set_open_position_to_closing(
+                        &mut conn,
+                        position.trader.to_string(),
+                        closing_price,
+                    )?;
+                    continue;
+                } else {
+                    tracing::trace!(trader_id, order_id, "Skipping liquidated position as match has already been found. Waiting for trader to come online to execute the trade.");
+                    continue;
+                }
+            }
+
+            tracing::info!(trader_id=%position.trader, ?best_current_price, %direction, position_id=%position.id, "Attempting to close liquidated position");
+
+            let new_order = NewMarketOrder {
+                id: uuid::Uuid::new_v4(),
+                contract_symbol: position.contract_symbol,
+                quantity: Decimal::try_from(position.quantity).expect("to fit into decimal"),
+                trader_id: position.trader,
+                direction: position.trader_direction.opposite(),
+                leverage: Decimal::from_f32(position.trader_leverage).expect("to fit into decimal"),
+                // This order can basically not expire, but if the user does not come back online
+                // within a certain time period we can assume the channel to be
+                // abandoned and we should force close.
+                expiry: OffsetDateTime::now_utc().add(LIQUIDATION_POSITION_TIMEOUT),
+                stable: position.stable,
+            };
+
+            let order = match orders::insert_market_order(
+                &mut conn,
+                new_order.clone(),
+                OrderReason::Liquidated,
+            ) {
+                Ok(order) => order,
+                Err(e) => {
+                    tracing::error!("Failed to insert liquidation order into DB. Error: {e:#}");
+                    continue;
+                }
+            };
+
+            let message = NewOrderMessage {
+                order,
+                channel_opening_params: None,
+                order_reason: OrderReason::Liquidated,
+            };
+
+            if let Err(e) = trading_sender.send(message).await {
+                tracing::error!(order_id=%new_order.id, trader_id=%new_order.trader_id, "Failed to submit new order for closing liquidated position. Error: {e:#}");
+                continue;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// Checks if the position needs to get liquidated. Either from the trader or the coordinator point
+/// of view.
+fn check_if_position_needs_to_get_liquidated(
```

Comment:
> Also we shouldn't need to recalculate the liquidation_price inside the function. We should have this in the position right?

yes, this has been addressed here 46eaa476cf0732d855f355b5a37243d04ecb63a9

Reply:
> 🔧 I find this function a bit too complex as it's checking both parties at the same time and returns a direction in which it's getting liquidated or None.
> 
> What's the reason for not having a function which only accepts one liquidation price?
> 
> It could look like this:
> 
> ```rust
> fn check_if_position_needs_to_get_liquidated(
>     trader_direction: Direction,
>     best_current_price: &Price,
>     liquidation_price: Decimal,
> ) -> bool {
> ...
> }
> ```
> 
> you can now call this twice, once for the coordinator and once for the trader.
> 
> The length of this function would then only be half what it's doing right now.

I refactored it quite significantly. Have a look here 0f1ca45f2371ae4711c0a4d97e0bd59e8ec92bf9